### PR TITLE
Own channels-last layout during backend construction

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary

Fixes the JetPack 6 ARM64 Docker regression introduced by #25999 and removes the duplicated channels-last mutation path introduced by #25983.

The old implementation constructed `AutoBackend`, then let each caller mutate its retained model through `set_memory_format()`. Automatic CPU enablement widened that path to every x86 CPU inference call, while model construction, fusion, and YOLOE setup could cross inference-mode boundaries and retain inference tensors.

This change restores the safe `False` default and makes backend construction the single owner of inference layout and retained tensor state:

- `BasePredictor` and `BaseValidator` pass the requested layout into `AutoBackend` instead of mutating a constructed backend.
- `AutoBackend` normalizes supplied inference-tensor modules before fusion, then applies layout once to native PyTorch models.
- Explicit channels-last remains supported on CUDA and PyTorch 1.13+ x86 CPUs with oneDNN.
- The automatic CPU policy, `set_memory_format()` helper, and redundant YOLOE outer inference decorator are deleted.
- The false branch of `smart_inference_mode` returns to its original direct composition.

The live diff is net-negative: 47 additions and 55 deletions.

## Validation

- Reproduced the production failure as 45 inference + 200 normal retained parameters.
- Verified the shared-owner repair retains 245 normal + 0 inference parameters, including modules supplied from caller inference mode.
- `python3 -m pytest tests/test_python.py::test_autobackend_memory_format -v`
- `python3 -m pytest tests/test_python.py::test_yoloe -v`
- `python3 -m pytest tests/test_python.py -k 'test_track_stream or test_predict_visualize or test_yolov10' -v`
- YOLOE standalone validation invoked inside an existing `torch.inference_mode()` context.
- Explicit `channels_last=False/True` public inference across detect, segment, semantic, depth, classify, pose, and OBB.
- Ruff format/lint, Prettier 3.8.5, and `git diff --check`.

Production failure: https://github.com/ultralytics/ultralytics/actions/runs/33309242381/job/99251174647

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary

`AutoBackend` now owns inference tensor normalization and channels-last layout selection, restoring `channels_last=False` by default and preventing retained inference tensors during backend construction and fusion.

### 📊 Key Changes

- `BasePredictor` and `BaseValidator` pass `channels_last` directly to `AutoBackend` and no longer mutate the constructed backend afterward.
- `AutoBackend` deep-copies native PyTorch modules containing inference tensors before fusion or other mutations, then applies the requested memory format once.
- Explicit `channels_last=True` is supported for native PyTorch models on CUDA and oneDNN-enabled x86 CPUs with PyTorch 1.13 or newer; unsupported devices and exported formats issue warnings and retain contiguous layout.
- The automatic CPU channels-last policy and `set_memory_format()` helper were removed, and the default is now `False` in configuration and prediction, training, and validation documentation.
- YOLOE validation no longer adds a redundant inference-mode decorator, while `smart_inference_mode(False)` returns to direct decorator composition.

### 🎯 Purpose & Impact

- Prevents backend construction and fusion from retaining inference tensors, including when modules are supplied from an existing inference-mode context.
- Removes automatic channels-last activation on eligible x86 CPU inference, so users must explicitly set `channels_last=True`.
- Keeps explicit layout selection centralized in `AutoBackend` and avoids applying it to unsupported backends such as ARM64, MPS, ONNX, and TensorRT.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/models/yolo/yoloe/val.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def get_vpe_dataloader(self, data: dict[str, Any]) -> torch.utils.data.DataLoade
             device=self.device,
         )
 
-    @smart_inference_mode()
+    @torch.no_grad()
```
Comment (UltralyticsAssistant) [inline ultralytics/models/yolo/yoloe/val.py]: 💡 **MEDIUM**: `torch.no_grad()` does not disable an already-active `torch.inference_mode()` context. Therefore standalone validation invoked from `with torch.inference_mode():` still performs the YOLOE setup and `deepcopy` with inference tensors, allowing the same `AutoBackend` fusion failure this change is intended to prevent. Use the existing `@smart_inference_mode(False)` helper, which explicitly exits inference mode while retaining no-grad behavior.

**Suggested change:**
```suggestion
    @smart_inference_mode(False)
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed the YOLOE validator decorator change and its BaseValidator/AutoBackend call path. The normal validation path is corrected, but callers already inside inference mode can still reproduce the mixed/inference-tensor fusion failure; the existing disabling helper should be used instead.

<details><summary>💬 Posted 1 inline comment</summary>

- 💡 **MEDIUM** `ultralytics/models/yolo/yoloe/val.py:134` `torch.no_grad()` does not disable an already-active `torch.inference_mode()` context. Therefore standalone validation invoked from `with torch.inference_mode():` still performs the YOLOE setup and `deepcopy` with inference tensors, allowing the same `AutoBackend` fusion failure this change is intended to prevent. Use the existing `@smart_inference_mode(False)` helper, which explicitly exits inference mode while retaining no-grad behavior.
</details>

File: ultralytics/nn/autobackend.py [hunk 2]
```diff
@@ -231,6 +233,23 @@ def __init__(
...
+            supported = device_type == "cuda" or (
+                TORCH_1_13
+                and device_type == "cpu"
+                and not ARM64
+                and torch.backends.mkldnn.is_available()
+                and torch.backends.mkldnn.enabled
+            )
+            if channels_last and not supported:
+                LOGGER.warning(f"'channels_last=True' is not supported on '{device_type}', ignoring.")
+            self.backend.model.to(
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/autobackend.py]: 💡 **MEDIUM**: This removes the inference-aware conversion needed when the supplied module already contains inference tensors. `smart_inference_mode(False)` disables the active context but does not convert existing parameters back to normal tensors, and `nn.Module.to(memory_format=...)` can fail while rewriting them outside `InferenceMode`. The deleted `set_memory_format()` explicitly detected `tensor.is_inference()` and ran `model.to` under `torch.inference_mode()`. Preserve that handling here for both memory-format branches; otherwise a retained model created under `torch.inference_mode()` can still fail during backend construction.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Since review 1, the YOLOE validator now uses `smart_inference_mode(False)`, addressing the previously reported standalone validation failure inside an existing inference-mode context. The backend ownership changes are otherwise coherent, but the removed inference-tensor-aware conversion leaves a realistic backend-construction regression when a retained module already contains inference tensors.

<details><summary>💬 Posted 1 inline comment</summary>

- 💡 **MEDIUM** `ultralytics/nn/autobackend.py:247` This removes the inference-aware conversion needed when the supplied module already contains inference tensors. `smart_inference_mode(False)` disables the active context but does not convert existing parameters back to normal tensors, and `nn.Module.to(memory_format=...)` can fail while rewriting them outside `InferenceMode`. The deleted `set_memory_format()` explicitly detected `tensor.is_inference()` and ran `model.to` under `torch.inference_mode()`. Preserve that handling here for both memor…
</details>

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 3

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Since review 2, the inference-tensor regression is addressed by normalizing supplied modules through a deepcopy before backend fusion and layout conversion, and the YOLOE decorator cleanup is consistent with AutoBackend now owning retained inference state. Reviewed all 10 changed files and relevant callers/tests; no remaining concrete correctness or compatibility issues found. LGTM.

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review 4

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Since review 3, the final backend-format cleanup and delegated-model regression test remain consistent with the ownership change. Reviewed all 10 changed files and relevant callers, backend implementations, inference-mode handling, and tests; no new concrete correctness, compatibility, security, or performance issues found. LGTM.

**Comment by UltralyticsAssistant** [comment]
🎉 [PR #26001](https://github.com/ultralytics/ultralytics/pull/26001) has been merged—thank you, @glenn-jocher!

> “Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away.” — Antoine de Saint-Exupéry

This is reflected in the streamlined `AutoBackend` design: inference tensor normalization and explicit channels-last selection are centralized, automatic layout changes were removed, and retained inference tensors are prevented during backend construction and fusion.
